docs/pages/machine-id/troubleshooting.mdx

```diff
@@ -0,0 +1,286 @@
+locked bot:
+ 1. Remove the lock on the bot's user
+ 2. Reset the bot's generation counter by deleting and re-creating the bot
+
+To remove the lock, first find and remove the lock targeting the bot user:
+
+```code
+$ tctl get locks
+kind: lock
+metadata:
+  id: 1658359514703080513
+  name: 5cee949f-5203-4f3b-9805-dac35d798a16
+spec:
+  message: The bot user "bot-example" has been locked due to a certificate generation
+    mismatch, possibly indicating a stolen certificate.
+  target:
+    user: bot-example
+version: v2
+
+$ tctl rm lock/5cee949f-5203-4f3b-9805-dac35d798a16
+```
+
+Next, reset the generation counter by deleting and recreating the bot:
+```code
+$ tctl bots rm example
+
+$ tctl bots add example --roles=foo,bar
+```
+
+Finally, reconfigure the bot with the new token and restart it. It will detect
+the new token and automatically reset its internal data directory.
+
+## `tbot` shows a "bad certificate error" at startup
+
+### Symptoms
+
+Restarting a `tbot` process outputs a log like the following:
+
+```text
+INFO [TBOT]      Successfully loaded bot identity, valid: after=2022-07-21T21:49:26Z, before=2022-07-21T22:50:26Z, duration=1h1m0s | kind=tls, renewable=true, disallow-reissue=false, roles=[bot-test], principals=[-teleport-internal-join], generation=2 tbot/tbot.go:281
+ERRO [TBOT]      Identity has expired. The renewal is likely to fail. (expires: 2022-07-21T22:50:26Z, current time: 2022-07-25T20:18:33Z) tbot/tbot.go:415
+WARN [TBOT]      Note: onboarding config ignored as identity was loaded from persistent storage tbot/tbot.go:288
+ERRO [TBOT]      Failed to resolve tunnel address Get "https://auth.example.com:3025/webapi/find": x509: cannot validate certificate for auth.example.com because it doesn't contain any IP SANs reversetunnel/transport.go:90
+ERRO [TBOT]      Failed to resolve tunnel address Get "https://auth.example.com:3025/webapi/find": x509: cannot validate certificate for auth.example.com because it doesn't contain any IP SANs reversetunnel/transport.go:90
+ERROR: failed direct dial to auth server: Get "https://teleport.cluster.local/v2/configuration/name": remote error: tls: bad certificate
+"\tGet \"https://teleport.cluster.local/v2/configuration/name\": remote error: tls: bad certificate, failed dial to auth server through reverse tunnel: Get \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://auth.example.com:3025/webapi/find\": x509: cannot validate certificate for auth.example.com because it doesn't contain any IP SANs"
+"\tGet \"https://teleport.cluster.local/v2/configuration/name\": Get \"https://auth.example.com:3025/webapi/find\": x509: cannot validate certificate for auth.example.com because it doesn't contain any IP SANs"
+```
+
+In particular, note the log line: "Identity has expired. The renewal is likely to
+fail."
+
+### Explanation
+
+Token-joined bots are unable to reauthenticate to the Teleport server once their
```

Comment:
Which Teleport server does this refer to?

Reply:
I'll clarify this to be the Teleport Auth Service. My only qualm (?) is that we'll also happily connect to the Proxy Service as well, but I don't think that'll really be a source of confusion..